src/language-server/project/rover/project.ts

```diff
@@ -38,6 +40,10 @@ export interface RoverProjectConfig extends GraphQLProjectConfig {
   capabilities: ClientCapabilities;
 }
 
+const supportedLanguageIds = (
+  process.env.APOLLO_ROVER_LANUGAGE_IDS || "graphql"
+).split(",");
```

Comment:
This resricts the rover LSP to only `.graphql`/`.gql`/etc. files and disables it for `.js` files and similar for now until we have a good user story there.